server/services/message-service.js

```diff
@@ -1,38 +1,56 @@
-const amqp = require('amqplib/callback_api')
-const scheduleService = require('./schedule-service')
+const rheaPromise = require('rhea-promise')
 const config = require('../config')
+const scheduleService = require('./schedule-service')
 
 module.exports = {
-  receiveSchedule: function () {
-    const messageQueue = config.messageQueue
-    amqp.connect(messageQueue, function (err, conn) {
-      if (err) {
-        console.log(err)
-      } else {
-        conn.createChannel(function (err, ch) {
-          if (err) {
-            console.log(err)
-          } else {
-            const scheduleQueue = 'schedule'
-            ch.assertQueue(scheduleQueue, { durable: false })
-            console.log('waiting for new claims')
+  setupReceivers: async function (messageQueueOptions) {
+    const connectionOptions = configureMQ(messageQueueOptions || config.messageQueue)
+    const connection = new rheaPromise.Connection(connectionOptions)
 
-            ch.consume(scheduleQueue, function (msg) {
-              console.log(`claim received for scheduling - ${msg.content.toString()}`)
-              scheduleService.create(JSON.parse(msg.content))
-            }, { noAck: true })
+    try {
+      await connection.open()
+    } catch (err) {

```

Comment:
Yes correct, it needs to stay open when the application is running as it doesn't know when it's going to receive messages.  I'll do as you suggest so it shuts down gracefully along with the application.

Reply:
I've added a SIGTERM listener to the message-service.js to shut down the connection.  I've tested it and it works but can you give me some feedback on how it can be improved?

```
process.on('SIGTERM', async function () {
      console.log('closing connection')
      await connection.close()
      process.exit(0)
    })
```